src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -1563,7 +1563,7 @@ public Location getLocation() {
     }
 
     public boolean isInsideOfWater() {
-        double y = this.y + this.getEyeHeight();
+        double y = this.y + this.getEyeHeight() - (this.isSprinting() ? 1 : 0);
```

Comment:
Surely it would make more sense to add this code inside of `getEyeHeight()` and use `isSwimming()` ?

Reply:
Well maybe in the Player class, but the getEyeHeight  refers to the Entity, so it might bug with mobs.

Reply:
Nevermind, still seems to work with mobs.

Reply:
You need to use `isSwimming()` instead of `isSprinting()`